src/main/java/spark/parser/commands/commandtypes/Command.java

```diff
@@ -0,0 +1,15 @@
+package spark.parser.commands.commandtypes;
+
+import spark.Ui;
+import spark.storage.Storage;
+import spark.tasks.TaskList;
+
+/**
+ * Represents a command that can be executed by Spark.
+ * All valid commands are sub-classes of this class.
+ */
+public abstract class Command {
+    public abstract String execute(TaskList tasks, Ui ui, Storage storage);
+
+    public abstract boolean isExit();
```

Comment:
Instead of having to override the `isExit` method in all the `Command` subclasses, you can consider using the java `instanceof` method since you do have an ExitCommand class to check if the command was to exit the application 😸